### ISSUE TITLE
Database Driver Know Before You Go To Avoid Discrepancy

### ISSUE THREAD

**Issue by chris206** [206Tours/DB#2026]
From: Chris Wright 
Subject: Ideas for a Database Driven KBYG 

Hey Sandy,

I was talking with Ronen today and Patrick previously about how we are re-doing our KBYG. Ronen mentioned that the KBYG should be database driven, and we both agreed.
We also agreed that the idea of having 50 different files would be an unmanageable mess; rife with discrepancies which would require that we re-engineer it again in no time. 

If we have 50 different sources, info will be almost instantly inconsistent.

I wanted to give you my idea to see what you think:

2 Tiered System:

1. Generic KBYG info level (50% of info is identical, so it should have the same source)
- How to prepare,
- tsa info
- meal info
- baggage
- general weather
- connections
- tipping info (without amounts)
- weather info (without amounts)
- certain clothing

2. Country specific info level (50% of info is based on country, so each should have same source)
- Real time currency conversion based on the IP address of the viewer
- weather average table for each country (or we could import info based on departure)
- Clothing to bring for each season in a table (or we could import it based on departure)
- VAT specifics
-Specific tip amounts
- Meal specific (such as lunches in Holy Land)
- Languages spoken
- Embassy info
- Power conversion

We would combine countries based on how they are selected in the Tour Code – with no need to actually specify a URL. They would spontaneously compile.

For example - if Holy Land & Rome are selected for Tour 114, each module would overlay the generic and look like a single document – even though they have 4 distinct sources.

You can visualize it like this (yellow is taken from generic source, green from Italy source, and red from Holy Land Source):

Language:
The official language of Italy is Italian.  English is widely spoken, and many signs appear in English too.
The official language of Israel is Hebrew and Arabic. English is widely spoken, and many signs appear in English too.
Please let me know if you have any questions. God Bless!

________________________________________
 	Chris Wright

**Comment by bradatkins** [comment]
Yup makes sense. Ronen will need to come up with a layout if there already isn't one.

**Comment by chris206** [comment]
So, this is out old layout. The new layout will be different. For purposes of helping everybody understand what I want to do.
The Tier 1 Lowest Common denominator stuff would be written in a narrative format - like a skeletal system
The Tier 2 country specific stuff could be added from a table of data cells. Wherever you see blue is the skeletal system, wherever you see yellow is imported data

![image](https://cloud.githubusercontent.com/assets/22382144/24616152/480b3554-185e-11e7-8ad9-707c9348fc09.png)
